server/conn.go

```diff
@@ -1062,7 +1062,7 @@ func (cc *clientConn) handleFieldList(sql string) (err error) {
 	if err != nil {
 		return errors.Trace(err)
 	}
-	data := make([]byte, 4, 1024)
+	data := cc.alloc.AllocWithLen(4, 1024)
```

Comment:
do we need to return the allocated memory buffer for reusing?

Reply:
It is reset after each finish a SQL statement, See https://github.com/pingcap/tidb/blob/fb94c3a56fc84341dd7c5452621d4ef91d1a6791/server/conn.go#L559